File: src/SendereCommons/SimpleNetworkInterface.java
package SendereCommons;

import lombok.Data;

import java.io.IOException;
import java.io.OutputStream;
import java.net.InterfaceAddress;
import java.net.NetworkInterface;
import java.net.SocketException;
import java.util.List;
import java.util.concurrent.ArrayBlockingQueue;
import java.util.concurrent.BlockingQueue;

@Data
public class SimpleNetworkInterface {

    public final String stringAddress;
    public final byte[] byteAddress;
    public final int subnetPrefixLength;
    public final byte[] macAddress;
    public final String systemName;
    public final String displayName;
    private final NetworkInterface networkInterface;

    private Thread senderThread;
    private Thread receiverThread;

    private Object senderLock = new Object();
    private Object receiverLock = new Object();

    private RemoteUser currentActiveUser = null;


    public SimpleNetworkInterface(NetworkInterface networkInterface) {
        this.networkInterface = networkInterface;
        systemName = networkInterface.getName();
        displayName = networkInterface.getDisplayName();
        byte[] tempMacAddress;
        try {
            tempMacAddress = networkInterface.getHardwareAddress();
        } catch (SocketException e) {
            tempMacAddress = new byte[]{0};
        }
        macAddress = tempMacAddress;
        String tempStringAddress = "null";
        byte[] tempByteAddress = new byte[]{0};
        int tempPrefixLength = 0;
        List<InterfaceAddress> addresses = networkInterface.getInterfaceAddresses();
        for (InterfaceAddress address : addresses) {
            if (address.getAddress().getHostAddress().contains(".")) {
                tempStringAddress = address.getAddress().getHostAddress();
                tempByteAddress = address.getAddress().getAddress();
                tempPrefixLength = address.getNetworkPrefixLength();
            }
        }
        subnetPrefixLength = tempPrefixLength;
        stringAddress = tempStringAddress;
        byteAddress = tempByteAddress;

    }
}

File: src/sendereCommons/SimpleNetworkInterface.kt
package sendereCommons

import lombok.Data
import java.io.IOException
import java.io.OutputStream
import java.lang.Exception
import java.lang.NullPointerException
import java.lang.Runnable
import java.net.NetworkInterface
import java.net.SocketException
import java.util.concurrent.ArrayBlockingQueue
import java.util.concurrent.BlockingQueue

@Data
class SimpleNetworkInterface(private val networkInterface: NetworkInterface) {
    val stringAddress: String
    val byteAddress: ByteArray
    val subnetPrefixLength: Int
    val macAddress: ByteArray
    val systemName: String
    val displayName: String

    init {
        systemName = networkInterface.name
        displayName = networkInterface.displayName
        macAddress = try {
            networkInterface.hardwareAddress
        } catch (e: Exception){
            byteArrayOf(0)
        }
        var tempStringAddress = "null"
        var tempByteAddress = byteArrayOf(0)
        var tempPrefixLength = 0
        val addresses = networkInterface.interfaceAddresses
        for (address in addresses) {
            if (address.address.hostAddress.contains(".")) {
                tempStringAddress = address.address.hostAddress
                tempByteAddress = address.address.address
                tempPrefixLength = address.networkPrefixLength.toInt()
            }
        }
        subnetPrefixLength = tempPrefixLength
        stringAddress = tempStringAddress
        byteAddress = tempByteAddress
    }
}